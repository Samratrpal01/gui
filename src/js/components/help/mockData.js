export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
