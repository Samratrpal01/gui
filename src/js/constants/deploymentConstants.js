const { SORTING_OPTIONS } = require('./appConstants');
const { DEVICE_LIST_DEFAULTS } = require('./deviceConstants');

const alreadyInstalled = 'already-installed';

const deploymentSubstates = {
  aborted: 'aborted',
  alreadyInstalled,
  decommissioned: 'decommissioned',
  downloading: 'downloading',
  failure: 'failure',
  installing: 'installing',
  noartifact: 'noartifact',
  pause_before_committing: 'pause_before_committing',
  pause_before_installing: 'pause_before_installing',
  pause_before_rebooting: 'pause_before_rebooting',
  pending: 'pending',
  rebooting: 'rebooting',
  success: 'success'
};

const deploymentStatesToSubstates = {
  failures: [deploymentSubstates.failure, deploymentSubstates.aborted, deploymentSubstates.decommissioned],
  inprogress: [deploymentSubstates.downloading, deploymentSubstates.installing, deploymentSubstates.rebooting],
  paused: [deploymentSubstates.pause_before_installing, deploymentSubstates.pause_before_rebooting, deploymentSubstates.pause_before_committing],
  pending: [deploymentSubstates.pending],
  successes: [deploymentSubstates.success, deploymentSubstates.alreadyInstalled, deploymentSubstates.noartifact]
};
const installationSubstatesMap = {
  download: {
    title: 'download',
    done: 'downloaded',
    successIndicators: [deploymentSubstates.installing, deploymentSubstates.rebooting, ...deploymentStatesToSubstates.paused, deploymentSubstates.success],
    failureIndicators: deploymentStatesToSubstates.failures,
    pauseConfigurationIndicator: 'ArtifactInstall_Enter'
  },
  install: {
    title: 'install',
    done: 'installed',
    successIndicators: [
      deploymentSubstates.rebooting,
      deploymentSubstates.pause_before_rebooting,
      deploymentSubstates.pause_before_committing,
      deploymentSubstates.success
    ],
    failureIndicators: deploymentStatesToSubstates.failures,
    pauseConfigurationIndicator: 'ArtifactReboot_Enter'
  },
  reboot: {
    title: 'reboot',
    done: 'rebooted',
    successIndicators: [deploymentSubstates.pause_before_committing, deploymentSubstates.success],
    failureIndicators: deploymentStatesToSubstates.failures,
    pauseConfigurationIndicator: 'ArtifactCommit_Enter'
  },
  commit: {
    title: 'commit',
    done: 'committed',
    successIndicators: deploymentStatesToSubstates.successes,
    failureIndicators: deploymentStatesToSubstates.failures,
    pauseConfigurationIndicator: undefined
  }
};

const DEPLOYMENT_STATES = {
  finished: 'finished',
  inprogress: 'inprogress',
  pending: 'pending',
  scheduled: 'scheduled'
};

const listDefaultsByState = {
  [DEPLOYMENT_STATES.inprogress]: { page: 1, perPage: 10 },
  [DEPLOYMENT_STATES.pending]: { page: 1, perPage: 10 },
  [DEPLOYMENT_STATES.scheduled]: { ...DEVICE_LIST_DEFAULTS },
  [DEPLOYMENT_STATES.finished]: { ...DEVICE_LIST_DEFAULTS },
  sort: { direction: SORTING_OPTIONS.desc }
};

module.exports = {
  CREATE_DEPLOYMENT: 'CREATE_DEPLOYMENT',
  REMOVE_DEPLOYMENT: 'REMOVE_DEPLOYMENT',
  RECEIVE_DEPLOYMENT: 'RECEIVE_DEPLOYMENT',
  RECEIVE_DEPLOYMENT_DEVICE_LOG: 'RECEIVE_DEPLOYMENT_DEVICE_LOG',
  RECEIVE_DEPLOYMENT_DEVICES: 'RECEIVE_DEPLOYMENT_DEVICES',
  RECEIVE_DEPLOYMENTS: 'RECEIVE_DEPLOYMENTS',
  RECEIVE_PENDING_DEPLOYMENTS: 'RECEIVE_PENDING_DEPLOYMENTS',
  RECEIVE_INPROGRESS_DEPLOYMENTS: 'RECEIVE_INPROGRESS_DEPLOYMENTS',
  RECEIVE_SCHEDULED_DEPLOYMENTS: 'RECEIVE_SCHEDULED_DEPLOYMENTS',
  RECEIVE_FINISHED_DEPLOYMENTS: 'RECEIVE_FINISHED_DEPLOYMENTS',
  SELECT_INPROGRESS_DEPLOYMENTS: 'SELECT_INPROGRESS_DEPLOYMENTS',
  SELECT_PENDING_DEPLOYMENTS: 'SELECT_PENDING_DEPLOYMENTS',
  SELECT_SCHEDULED_DEPLOYMENTS: 'SELECT_SCHEDULED_DEPLOYMENTS',
  SELECT_FINISHED_DEPLOYMENTS: 'SELECT_FINISHED_DEPLOYMENTS',
  SELECT_DEPLOYMENT: 'SELECT_DEPLOYMENT',
  SET_DEPLOYMENTS_STATE: 'SET_DEPLOYMENTS_STATE',
  DEFAULT_PENDING_INPROGRESS_COUNT: 10,
  DEPLOYMENT_ROUTES: {
    active: {
      key: 'active',
      route: '/deployments/active',
      title: 'Active'
    },
    finished: {
      key: 'finished',
      route: '/deployments/finished',
      title: 'Finished'
    },
    scheduled: {
      key: 'scheduled',
      route: '/deployments/scheduled',
      title: 'Scheduled'
    }
  },
  DEPLOYMENT_STATES,
  DEPLOYMENT_TYPES: {
    software: 'software',
    configuration: 'configuration'
  },
  defaultStats: {
    [deploymentSubstates.aborted]: 0,
    [deploymentSubstates.alreadyInstalled]: 0,
    [deploymentSubstates.decommissioned]: 0,
    [deploymentSubstates.downloading]: 0,
    [deploymentSubstates.failure]: 0,
    [deploymentSubstates.installing]: 0,
    [deploymentSubstates.noartifact]: 0,
    [deploymentSubstates.pause_before_committing]: 0,
    [deploymentSubstates.pause_before_installing]: 0,
    [deploymentSubstates.pause_before_rebooting]: 0,
    [deploymentSubstates.pending]: 0,
    [deploymentSubstates.rebooting]: 0,
    [deploymentSubstates.success]: 0
  },
  deploymentDisplayStates: {
    finished: 'Finished',
    scheduled: 'Scheduled',
    skipped: 'Skipped',
    paused: 'Paused',
    pending: 'Pending',
    inprogress: 'In Progress',
    success: 'Success',
    successes: 'Success',
    failure: 'Fail',
    failures: 'Fail'
  },
  deploymentStatesToSubstates,
  deploymentStatesToSubstatesWithSkipped: {
    ...deploymentStatesToSubstates,
    failures: [deploymentSubstates.failure],
    skipped: [deploymentSubstates.aborted, deploymentSubstates.noartifact, deploymentSubstates.alreadyInstalled, deploymentSubstates.decommissioned],
    successes: [deploymentSubstates.success]
  },
  deploymentSubstates,
  deploymentPrototype: {
    devices: {},
    name: undefined,
    stats: {}
  },
  installationSubstatesMap,
  listDefaultsByState,
  pauseMap: {
    [deploymentSubstates.pause_before_installing]: {
      title: installationSubstatesMap.download.done,
      followUp: installationSubstatesMap.download.pauseConfigurationIndicator
    },
    [deploymentSubstates.pause_before_rebooting]: {
      title: installationSubstatesMap.install.done,
      followUp: installationSubstatesMap.install.pauseConfigurationIndicator
    },
    [deploymentSubstates.pause_before_committing]: {
      title: installationSubstatesMap.reboot.done,
      followUp: installationSubstatesMap.reboot.pauseConfigurationIndicator
    }
  }
};
