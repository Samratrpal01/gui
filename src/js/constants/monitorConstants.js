module.exports = {
  CHANGE_ALERT_CHANNEL: 'CHANGE_ALERT_CHANNEL',
  RECEIVE_DEVICE_ALERTS: 'RECEIVE_DEVICE_ALERTS',
  RECEIVE_DEVICE_MONITOR_CONFIG: 'RECEIVE_DEVICE_MONITOR_CONFIG',
  RECEIVE_DEVICE_ISSUE_COUNTS: 'RECEIVE_DEVICE_ISSUE_COUNTS',
  RECEIVE_LATEST_DEVICE_ALERTS: 'RECEIVE_LATEST_DEVICE_ALERTS',
  alertChannels: {
    email: 'email'
  }
};
