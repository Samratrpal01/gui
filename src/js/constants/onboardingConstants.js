export const onboardingSteps = {
  ONBOARDING_START: 'onboarding-start',
  DEVICES_PENDING_ONBOARDING_START: 'devices-pending-onboarding-start',
  DASHBOARD_ONBOARDING_START: 'dashboard-onboarding-start',
  DASHBOARD_ONBOARDING_PENDINGS: 'dashboard-onboarding-pendings',
  DEVICES_PENDING_ONBOARDING: 'devices-pending-onboarding',
  DEVICES_ACCEPTED_ONBOARDING: 'devices-accepted-onboarding',
  DEVICES_PENDING_ACCEPTING_ONBOARDING: 'devices-pending-accepting-onboarding',
  DEVICES_ACCEPTED_ONBOARDING_NOTIFICATION: 'devices-accepted-onboarding-notification',
  UPLOAD_PREPARED_ARTIFACT_TIP: 'upload-prepared-artifact-tip',
  APPLICATION_UPDATE_REMINDER_TIP: 'application-update-reminder-tip',
  ARTIFACT_INCLUDED_ONBOARDING: 'artifact-included-onboarding',
  ARTIFACT_INCLUDED_DEPLOY_ONBOARDING: 'artifact-included-deploy-onboarding',
  SCHEDULING_ARTIFACT_SELECTION: 'scheduling-artifact-selection',
  SCHEDULING_ALL_DEVICES_SELECTION: 'scheduling-all-devices-selection',
  SCHEDULING_GROUP_SELECTION: 'scheduling-group-selection',
  SCHEDULING_RELEASE_TO_DEVICES: 'scheduling-release-to-devices',
  DEPLOYMENTS_INPROGRESS: 'deployments-inprogress',
  DEPLOYMENTS_PAST: 'deployments-past',
  DEPLOYMENTS_PAST_COMPLETED: 'deployments-past-completed',
  DEPLOYMENTS_PAST_COMPLETED_NOTIFICATION: 'deployments-past-completed-notification',
  DEPLOYMENTS_PAST_COMPLETED_FAILURE: 'deployments-past-completed-failure',
  ARTIFACT_CREATION_DIALOG: 'artifact-creation-dialog',
  UPLOAD_NEW_ARTIFACT_TIP: 'upload-new-artifact-tip',
  UPLOAD_NEW_ARTIFACT_DIALOG_UPLOAD: 'upload-new-artifact-dialog-upload',
  UPLOAD_NEW_ARTIFACT_DIALOG_DESTINATION: 'upload-new-artifact-dialog-destination',
  UPLOAD_NEW_ARTIFACT_DIALOG_DEVICE_TYPE: 'upload-new-artifact-dialog-device-type',
  UPLOAD_NEW_ARTIFACT_DIALOG_RELEASE_NAME: 'upload-new-artifact-dialog-release-name',
  ARTIFACT_MODIFIED_ONBOARDING: 'artifact-modified-onboarding',
  ONBOARDING_FINISHED: 'onboarding-finished',
  ONBOARDING_FINISHED_NOTIFICATION: 'onboarding-finished-notification',
  ONBOARDING_CANCELED: 'onboarding-canceled'
};
export const SET_DEMO_ARTIFACT_PORT = 'SET_DEMO_ARTIFACT_PORT';
export const SET_ONBOARDING_COMPLETE = 'SET_ONBOARDING_COMPLETE';
export const SET_ONBOARDING_PROGRESS = 'SET_ONBOARDING_PROGRESS';
export const SET_ONBOARDING_DEVICE_TYPE = 'SET_ONBOARDING_DEVICE_TYPE';
export const SET_ONBOARDING_APPROACH = 'SET_ONBOARDING_APPROACH';
export const SET_ONBOARDING_ARTIFACT_INCLUDED = 'SET_ONBOARDING_ARTIFACT_INCLUDED';
export const SET_SHOW_CREATE_ARTIFACT = 'SET_SHOW_CREATE_ARTIFACT';
export const SET_SHOW_ONBOARDING_HELP = 'SET_SHOW_ONBOARDING_HELP';
export const SET_SHOW_ONBOARDING_HELP_DIALOG = 'SET_SHOW_ONBOARDING_HELP_DIALOG';
