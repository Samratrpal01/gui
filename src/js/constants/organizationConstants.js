import { EXTERNAL_PROVIDER } from './deviceConstants';

export const AUDIT_LOGS_TYPES = [
  { title: 'Deployment', queryParameter: 'object_deployment_name', value: 'deployment' },
  { title: 'Device', queryParameter: 'object_id', value: 'device' },
  { title: 'User', queryParameter: 'object_id', value: 'user' }
];
export const RECEIVE_AUDIT_LOGS = 'RECEIVE_AUDIT_LOGS';
export const RECEIVE_CURRENT_CARD = 'RECEIVE_CURRENT_CARD';
export const RECEIVE_SETUP_INTENT = 'RECEIVE_SETUP_INTENT';
export const SET_AUDITLOG_STATE = 'SET_AUDITLOG_STATE';
export const SET_ORGANIZATION = 'SET_ORGANIZATION';
export const RECEIVE_EXTERNAL_DEVICE_INTEGRATIONS = 'RECEIVE_EXTERNAL_DEVICE_INTEGRATIONS';
export const RECEIVE_SAML_CONFIGS = 'RECEIVE_SAML_CONFIGS';
export const RECEIVE_WEBHOOK_EVENTS = 'RECEIVE_WEBHOOK_EVENTS';
export const emptyWebhook = {
  description: '',
  enabled: true,
  credentials: {
    type: EXTERNAL_PROVIDER.webhook.credentialsType,
    [EXTERNAL_PROVIDER.webhook.credentialsType]: {
      secret: '',
      url: ''
    }
  }
};
