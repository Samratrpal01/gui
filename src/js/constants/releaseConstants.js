const defaultVisibleSection = { start: 0, end: 0 };

module.exports = {
  ARTIFACT_GENERATION_TYPE: {
    SINGLE_FILE: 'single_file'
  },
  ARTIFACTS_REMOVED_ARTIFACT: 'ARTIFACTS_REMOVED_ARTIFACT',
  ARTIFACTS_SET_ARTIFACT_URL: 'ARTIFACTS_SET_ARTIFACT_URL',
  UPDATED_ARTIFACT: 'UPDATED_ARTIFACT',
  RECEIVE_ARTIFACTS: 'RECEIVE_ARTIFACTS',
  RECEIVE_RELEASE: 'RECEIVE_RELEASE',
  RECEIVE_RELEASES: 'RECEIVE_RELEASES',
  RELEASE_REMOVED: 'RELEASE_REMOVED',
  SHOW_REMOVE_DIALOG: 'SHOW_REMOVE_DIALOG',
  SELECTED_ARTIFACT: 'SELECTED_ARTIFACT',
  SELECTED_RELEASE: 'SELECTED_RELEASE',
  SET_RELEASES_LIST_STATE: 'SET_RELEASES_LIST_STATE',
  defaultVisibleSection,
  rootfsImageVersion: 'rootfs-image.version'
};
