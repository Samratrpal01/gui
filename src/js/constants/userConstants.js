const { apiUrl } = require('../api/general-api');

const useradmApiUrl = `${apiUrl.v1}/useradm`;

const staticRolesByName = {
  admin: 'RBAC_ROLE_PERMIT_ALL',
  readOnly: 'RBAC_ROLE_OBSERVER',
  ci: 'RBAC_ROLE_CI',
  deploymentsManager: 'RBAC_ROLE_DEPLOYMENTS_MANAGER',
  terminalAccess: 'RBAC_ROLE_REMOTE_TERMINAL'
};

module.exports = {
  RECEIVED_QR_CODE: 'RECEIVED_QR_CODE',

  SUCCESSFULLY_LOGGED_IN: 'SUCCESSFULLY_LOGGED_IN',
  USER_LOGOUT: 'USER_LOGOUT',
  RECEIVED_ACTIVATION_CODE: 'RECEIVED_ACTIVATION_CODE',
  RECEIVED_USER_LIST: 'RECEIVED_USER_LIST',
  RECEIVED_USER: 'RECEIVED_USER',
  CREATED_USER: 'CREATED_USER',
  REMOVED_USER: 'REMOVED_USER',
  UPDATED_USER: 'UPDATED_USER',

  RECEIVED_ROLES: 'RECEIVED_ROLES',
  CREATED_ROLE: 'CREATED_ROLE',
  UPDATED_ROLE: 'UPDATED_ROLE',
  REMOVED_ROLE: 'REMOVED_ROLE',

  SET_CUSTOM_COLUMNS: 'SET_CUSTOM_COLUMNS',
  SET_GLOBAL_SETTINGS: 'SET_GLOBAL_SETTINGS',
  SET_SHOW_HELP: 'SET_SHOW_HELP',
  SET_SHOW_CONNECT_DEVICE: 'SET_SHOW_CONNECT_DEVICE',

  OWN_USER_ID: 'me',
  emptyRole: { title: undefined, allowUserManagement: false, groups: [], description: '', editable: undefined, permissions: [] },
  rolesById: {
    [staticRolesByName.admin]: { title: 'Admin', allowUserManagement: true, groups: [], description: 'Full access', editable: false, permissions: [] },
    [staticRolesByName.readOnly]: { title: 'Read Access', allowUserManagement: false, groups: [], description: '', editable: false, permissions: [] },
    [staticRolesByName.ci]: { title: 'Releases Manager', allowUserManagement: false, groups: [], description: '', editable: false, permissions: [] },
    [staticRolesByName.deploymentsManager]: {
      title: 'Deployments Manager',
      allowUserManagement: false,
      groups: [],
      description: '',
      editable: false,
      permissions: []
    },
    [staticRolesByName.terminalAccess]: {
      title: 'Troubleshooting',
      allowUserManagement: false,
      groups: [],
      description: 'Access to the troubleshooting features: Remote Terminal, File Transfer, Port Forwarding',
      editable: false,
      permissions: []
    }
  },
  rolesByName: {
    ...staticRolesByName,
    userManagement: { action: 'http', object: { type: 'any', value: `${useradmApiUrl}/.*` } },
    deploymentCreation: { action: 'CREATE_DEPLOYMENT', object: { type: 'DEVICE_GROUP', value: undefined } },
    groupAccess: { action: 'VIEW_DEVICE', object: { type: 'DEVICE_GROUP', value: undefined } }
  },
  twoFAStates: {
    enabled: 'enabled',
    disabled: 'disabled',
    unverified: 'unverified'
  },
  settingsKeys: {
    initialized: 'settings-initialized'
  },
  useradmApiUrl
};
