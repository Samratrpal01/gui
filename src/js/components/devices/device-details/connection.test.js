import React from 'react';
import DeviceConnection, { DeviceConnectionMissingNote, DeviceDisconnectedNote, PortForwardLink } from './connection';
import { defaultState, undefineds } from '../../../../../tests/mockData';
import { render } from '../../../../../tests/setupTests';
import { DEVICE_CONNECT_STATES } from '../../../constants/deviceConstants';

describe('tiny DeviceConnection components', () => {
  [DeviceConnectionMissingNote, DeviceDisconnectedNote, PortForwardLink].forEach(async Component => {
    it(`renders ${Component.displayName || Component.name} correctly`, () => {
      const { baseElement } = render(<Component docsVersion="" lastConnectionTs={defaultState.devices.byId.a1.updated_ts} />);
      const view = baseElement.firstChild;
      expect(view).toMatchSnapshot();
      expect(view).toEqual(expect.not.stringMatching(undefineds));
    });
  });
});

describe('DeviceConnection Component', () => {
  const userRoles = { canTroubleshoot: true, hasWriteAccess: true };
  it('renders correctly', async () => {
    const { baseElement } = render(<DeviceConnection device={defaultState.devices.byId.a1} setSnackbar={jest.fn} userRoles={userRoles} />);
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
  it('renders correctly when disconnected', async () => {
    const { baseElement } = render(
      <DeviceConnection
        device={{ ...defaultState.devices.byId.a1, connect_status: DEVICE_CONNECT_STATES.disconnected }}
        setSnackbar={jest.fn}
        userRoles={userRoles}
      />
    );
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
  it('renders correctly when connected', async () => {
    const { baseElement } = render(
      <DeviceConnection
        device={{ ...defaultState.devices.byId.a1, connect_status: DEVICE_CONNECT_STATES.connected }}
        setSnackbar={jest.fn}
        userRoles={userRoles}
      />
    );
    const view = baseElement.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
