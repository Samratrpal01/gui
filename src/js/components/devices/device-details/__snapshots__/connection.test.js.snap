// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceConnection Component renders correctly 1`] = `
<div>
  <div
    class="margin-bottom "
  >
    <div
      class=" flexbox space-between"
      style="align-items: flex-start;"
    >
      <h4
        class="margin-bottom-small"
      >
        Troubleshoot
      </h4>
      <div
        class="flexbox centered"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
          disabled=""
          style="margin-right: 16px; font-weight: 900; font-size: smaller;"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Add-on
          </span>
        </button>
        <div
          style="height: 48px; width: 48px;"
        />
      </div>
    </div>
    <div>
      <div
        class="flexbox"
        style="flex-direction: row;"
      >
        <div
          class="flexbox text-muted"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            style="margin-right: 8px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1"
            style="text-transform: none; text-align: left;"
          >
            The troubleshoot add-on does not seem to be enabled on this device.
            <br />
            Please
             
            <a
              href="https://docs.mender.io/add-ons/remote-terminal"
              rel="noopener noreferrer"
              target="_blank"
            >
              see the documentation
            </a>
             
            for a description on how it works and how to enable it.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DeviceConnection Component renders correctly when connected 1`] = `
<div>
  <div
    class="margin-bottom "
  >
    <div
      class=" flexbox space-between"
      style="align-items: flex-start;"
    >
      <h4
        class="margin-bottom-small"
      >
        Troubleshoot
      </h4>
      <div
        class="flexbox centered"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
          disabled=""
          style="margin-right: 16px; font-weight: 900; font-size: smaller;"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Add-on
          </span>
        </button>
        <div
          style="height: 48px; width: 48px;"
        />
      </div>
    </div>
    <div>
      <div
        class="flexbox"
        style="flex-direction: row;"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
          disabled=""
          style="margin-right: 16px;"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20,19V7H4V19H20M20,3A2,2 0 0,1 22,5V19A2,2 0 0,1 20,21H4A2,2 0 0,1 2,19V5C2,3.89 2.9,3 4,3H20M13,17V15H18V17H13M9.58,13L5.57,9H8.4L11.7,12.3C12.09,12.69 12.09,13.33 11.7,13.72L8.42,17H5.59L9.58,13Z"
                />
              </svg>
            </span>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
              style="text-transform: none; text-align: left;"
            >
              Launch a new Remote Terminal session
            </h6>
          </span>
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
          disabled=""
          style="margin-right: 16px;"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 3L5 6.99h3V14h2V6.99h3L9 3zm7 14.01V10h-2v7.01h-3L15 21l4-3.99h-3z"
                />
              </svg>
            </span>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2"
              style="text-transform: none; text-align: left;"
            >
              Launch File Transfer
            </h6>
          </span>
        </button>
        <a
          class="flexbox centered margin-left"
          href="https://docs.mender.io/add-ons/port-forward"
          rel="noopener noreferrer"
          target="_blank"
        >
          Enable port forwarding
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root margin-left-small MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`DeviceConnection Component renders correctly when disconnected 1`] = `
<div>
  <div
    class="margin-bottom "
  >
    <div
      class=" flexbox space-between"
      style="align-items: flex-start;"
    >
      <h4
        class="margin-bottom-small"
      >
        Troubleshoot
      </h4>
      <div
        class="flexbox centered"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
          disabled=""
          style="margin-right: 16px; font-weight: 900; font-size: smaller;"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Add-on
          </span>
        </button>
        <div
          style="height: 48px; width: 48px;"
        />
      </div>
    </div>
    <div>
      <div
        class="flexbox"
        style="flex-direction: row;"
      >
        <div
          class="flexbox text-muted"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
            focusable="false"
            style="margin-right: 8px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1"
            style="text-transform: none; text-align: left;"
          >
            The troubleshoot add-on is not currently connected on this device, it was last connected on 
            <span>
              Invalid date
            </span>
            .
            <br />
            Please
             
            <a
              href="https://docs.mender.io/add-ons/remote-terminal"
              rel="noopener noreferrer"
              target="_blank"
            >
              see the documentation
            </a>
             
            for more information.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`tiny DeviceConnection components renders DeviceConnectionMissingNote correctly 1`] = `
<div>
  <div
    class="flexbox text-muted"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
      focusable="false"
      style="margin-right: 8px;"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
    <p
      class="MuiTypography-root MuiTypography-body1"
      style="text-transform: none; text-align: left;"
    >
      The troubleshoot add-on does not seem to be enabled on this device.
      <br />
      Please
       
      <a
        href="https://docs.mender.io/add-ons/remote-terminal"
        rel="noopener noreferrer"
        target="_blank"
      >
        see the documentation
      </a>
       
      for a description on how it works and how to enable it.
    </p>
  </div>
</div>
`;

exports[`tiny DeviceConnection components renders DeviceDisconnectedNote correctly 1`] = `
<div>
  <div
    class="flexbox text-muted"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
      focusable="false"
      style="margin-right: 8px;"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
      />
    </svg>
    <p
      class="MuiTypography-root MuiTypography-body1"
      style="text-transform: none; text-align: left;"
    >
      The troubleshoot add-on is not currently connected on this device, it was last connected on 
      <time
        datetime="2019-01-01T09:25:00+00:00"
      >
        2019-01-01 09:25
      </time>
      .
      <br />
      Please
       
      <a
        href="https://docs.mender.io/add-ons/remote-terminal"
        rel="noopener noreferrer"
        target="_blank"
      >
        see the documentation
      </a>
       
      for more information.
    </p>
  </div>
</div>
`;

exports[`tiny DeviceConnection components renders PortForwardLink correctly 1`] = `
<div>
  <a
    class="flexbox centered margin-left"
    href="https://docs.mender.io/add-ons/port-forward"
    rel="noopener noreferrer"
    target="_blank"
  >
    Enable port forwarding
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root margin-left-small MuiSvgIcon-fontSizeSmall"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
      />
    </svg>
  </a>
</div>
`;
